kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java

```diff
@@ -20,167 +18,191 @@
-	}
+public class KuraException extends Exception {
+
+    /** The Constant denoting resource bundle. */
+    private static final String KURA_EXCEPTION_MESSAGES_BUNDLE = "org.eclipse.kura.core.messages.KuraExceptionMessagesBundle";
+
+    /** The Constant denoting message pattern. */
+    private static final String KURA_GENERIC_MESSAGES_PATTERN = "Generic Error - {0}: {1}";
+
+    /** The Logger. */
+    private static final Logger s_logger = LoggerFactory.getLogger(KuraException.class);
+
+    /** The Constant denoting serial version identifier. */
+    private static final long serialVersionUID = 7468633737373095296L;
+
+    /** The associated arguments. */
+    private transient Object[] arguments;
+
+    /** The associated error code. */
+    private final KuraErrorCode code;
+
+    /**
+     * Builds a new {@link KuraException} instance based on the supplied {@link KuraErrorCode}.
+     *
+     * @param code
+     *            the error code
+     */
+    public KuraException(final KuraErrorCode code) {
+        this.code = code;
+    }
+
+    /**
+     * Builds a new {@link KuraException} instance based on the supplied {@link KuraErrorCode}.
+     *
+     * @param code
+     *            the error code
+     * @param arguments
+     *            the arguments
+     */
+    public KuraException(final KuraErrorCode code, final Object... arguments) {
+        this.code = code;
+        this.arguments = arguments;
+    }
+
+    /**
+     * Builds a new {@link KuraException} instance based on the supplied {@link KuraErrorCode}, an optional Throwable
+     * cause, and optional arguments for the associated exception message.
+     *
+     * @param code
+     *            the error code
+     * @param cause
+     *            the cause
+     * @param arguments
+     *            the arguments
+     */
+    public KuraException(final KuraErrorCode code, final Throwable cause, final Object... arguments) {
+        super(cause);
+        this.code = code;
+        this.arguments = arguments;
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing
+     * only a message.
+     *
+     * @param message
+     *            the message
+     * @return the kura exception
+     */
+    public static KuraException internalError(final String message) {
+        return new KuraException(KuraErrorCode.INTERNAL_ERROR, null, message);
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing a
+     * cause and a message.
+     *
+     * @param cause
+     *            the cause
+     * @return the kura exception
+     */
+    public static KuraException internalError(final Throwable cause) {
+        return new KuraException(KuraErrorCode.INTERNAL_ERROR, cause, "");
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing a
```

Comment:
There should be at least a comment to not use this one. As we agreed on not using "INTERNAL_ERROR" anymore.